cmd/lifecycle/analyzer.go

```diff
@@ -20,51 +21,71 @@ import (
 
 type analyzeCmd struct {
 	//flags: inputs
-	cacheDir      string
-	cacheImageTag string
-	groupPath     string
-	uid, gid      int
 	analyzeArgs
+	uid, gid int
+
+	cacheDir      string // Platform API < 0.7
+	cacheImageTag string // Platform API < 0.7
+	groupPath     string // Platform API < 0.7
 
 	//flags: paths to write data
 	analyzedPath string
 }
 
 type analyzeArgs struct {
 	//inputs needed when run by creator
-	imageName  string
-	layersDir  string
-	skipLayers bool
-	useDaemon  bool
-
-	platform cmd.Platform
-
-	//construct if necessary before dropping privileges
-	docker   client.CommonAPIClient
-	keychain authn.Keychain
+	imageName   string
+	layersDir   string
+	orderPath   string //nolint - Platform API >= 0.7
+	runImageRef string //nolint - Platform API >= 0.7
+	stackPath   string //nolint - Platform API >= 0.7
+	skipLayers  bool   // Platform API < 0.7
+	useDaemon   bool
+
+	additionalTags cmd.StringSlice        //nolint Platform API >= 0.7
+	cache          lifecycle.Cache        // Platform API < 0.7
+	docker         client.CommonAPIClient //construct if necessary before dropping privileges
+	group          buildpack.Group        // Platform API < 0.7
+	keychain       authn.Keychain
+	platform       cmd.Platform
```

Comment:
Now that we have all these fields that aren't technically provided as inputs, I wonder if it makes sense to rename this struct.

Reply:
`analyzeInputs?`

Also, maybe a re-organization of this sort would make things clearer:
```
type analyzeArgs struct {
	platform06Inputs
	platform07Inputs
	imageName string
	layersDir string
	useDaemon bool

	docker   client.CommonAPIClient
	keychain authn.Keychain
	platform cmd.Platform
}

type platform06Inputs struct {
	skipLayers bool
	cache      lifecycle.Cache
	group      buildpack.Group
}

type platform07Inputs struct {
	orderPath      string          //nolint
	runImageRef    string          //nolint
	stackPath      string          //nolint
	additionalTags cmd.StringSlice //nolint
}
```

Reply:
Hmm. I don't mind the `platform06Inputs`, but the `platform07Inputs` feels a bit odd to me. Unless we decide to undo everything in platform 0.7 one day, those fields will likely be built upon just like the previous 0.6 fields (layersDir, imageName, docker, etc.). We would likely only ever remove a field or two down the line.

How would you feel about something like 

```
type analyzeArgs struct {
  p06 platform06Inputs
}

...
cmd.FlagSkipLayers(&a.p06.skipLayers)

...

	analyzedMD, err := (&lifecycle.Analyzer{
		Buildpacks:    aa.p06.group.Group,
		Cache:         aa.p06.cache,
		Logger:        cmd.DefaultLogger,
		SkipLayers:    aa.p06.skipLayers,
		Platform:      aa.platform,
		Image:         img,
		LayerAnalyzer: lifecycle.NewLayerAnalyzer(cmd.DefaultLogger, cacheMetaRetriever, aa.layersDir, aa.platform),
	}).Analyze()

```

I kind of like how those legacy fields stand out - you would think twice before using one if you are developing a 0.7 or newer feature.